### PR TITLE
Cave-in walls, rails decals and the addition of a functioning mining wagon.

### PR DESCRIPTION
**Pull request by Blacknas**
## About The Pull Request
Creates cave-in walls, rail decals and a functioning mining wagon inside the following directory:
fulpstation\fulp_modules\maps\decals

## Why It's Good For The Game

Allows map makers to add depth to the station.

image
image
image
image
image /obj/vehicle/ridden/mining_wagon
image /obj/effect/turf_decal/rails
image /obj/effect/turf_decal/rails/vertical

The following 2 sprites are part of one sprite:
image /obj/effect/turf_decal/rails/broken
image

The following 4 sprites are part of one sprite:
image /obj/effect/turf_decal/rails/vertical_curved
image
image
image

image
Both cave-in sprites are part of /turf/closed/mineral/cave_in (which is a wall by itself, the sprites below are overlays)
image /turf/closed/mineral/cave_in
image /turf/closed/mineral/cave_in/blood

## Changelog
:cl:
add: Cave-in walls, rail decals and a functioning mining wagon.
/:cl:

### CODE REVIEW

**Comment by SpaceDragon00** [comment]
PR to Eos directly

**Comment by SpaceDragon00** [comment]
Also compounding my thoughts on the sprites here for future reference
- Tracks appear to be colored by the turf overlay (The tracks are significantly darker on the grey tile)
- Both tracks should probably be styled as the vertical ones, since the horizontal to vertical transition looks weird
- The cave-in decal isn't centered, and it's kinda hard to tell what it is

File: fulp_modules/maps/decals/mining_decals.dm [hunk 1]
```diff
@@ -0,0 +1,86 @@
+//MINING WAGON--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+/obj/vehicle/ridden/mining_wagon
+	name = "mining wagon"
+	desc = "Looks surprisingly new for a ancient mining wagon. The wheels seems to be stuck."
```
Comment (SpaceDragon00) [inline fulp_modules/maps/decals/mining_decals.dm:5]: an* ancient mining wagon
The wheels seem*

File: fulp_modules/maps/decals/mining_decals.dm [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	var/mutable_appearance/activated_overlay
+
+/turf/closed/mineral/cave_in/Initialize()
+	. = ..()
+	activated_overlay = mutable_appearance('fulp_modules/maps/decals/cave_in.dmi', "cave_in", ON_EDGED_TURF_LAYER)
+	add_overlay(activated_overlay)
+
+/turf/closed/mineral/cave_in/blood
+	name = "bloody ancient cave-in"
+	desc = "A ancient mine that collapsed which is housing equally ancient miners."
```
Comment (SpaceDragon00) [inline fulp_modules/maps/decals/mining_decals.dm:79]: An* ancient mine that collapsed

Comment (Blacknas) [reply fulp_modules/maps/decals/mining_decals.dm:79]: fixed the spelling issues, thank you for pointing them out

**Comment by Helianthus00** [comment]
This is interesting, but it really should be pr'd upstream at tg. Please do it there and if they reject it, you can return it here in a modular way.

**Pull request closed without merging**